runelite-client/src/main/java/net/runelite/client/plugins/config/PluginListItem.java

```diff
@@ -0,0 +1,300 @@
+	private static final ImageIcon ON_SWITCHER;
+	private static final ImageIcon OFF_SWITCHER;
+	private static final ImageIcon ON_STAR;
+	private static final ImageIcon OFF_STAR;
+
+	private final ConfigPanel configPanel;
+
+	@Getter
+	@Nullable
+	private final Plugin plugin;
+
+	@Nullable
+	private final Config config;
+
+	@Nullable
+	private final ConfigDescriptor configDescriptor;
+
+	@Getter
+	private final String name;
+
+	@Getter
+	private final String description;
+
+	private final List<String> keywords = new ArrayList<>();
+
+	private final JLabel pinButton = new JLabel(OFF_STAR, SwingConstants.LEFT);
+	private final JLabel configButton = new JLabel(CONFIG_ICON);
+	private final JLabel toggleButton = new JLabel(OFF_SWITCHER, SwingConstants.RIGHT);
+
+	private boolean isPluginEnabled = false;
+
+	@Getter
+	private boolean isPinned = false;
+
+	static
+	{
+		try
+		{
+			synchronized (ImageIO.class)
+			{
+				BufferedImage configIcon = ImageIO.read(ConfigPanel.class.getResourceAsStream("config_edit_icon.png"));
+				CONFIG_ICON = new ImageIcon(configIcon);
+				CONFIG_ICON_HOVER = new ImageIcon(SwingUtil.grayscaleOffset(configIcon, -100));
+				ON_SWITCHER = new ImageIcon(ImageIO.read(ConfigPanel.class.getResourceAsStream("switchers/on.png")));
+				OFF_SWITCHER = new ImageIcon(ImageIO.read(ConfigPanel.class.getResourceAsStream("switchers/off.png")));
+				ON_STAR = new ImageIcon(ImageIO.read(ConfigPanel.class.getResourceAsStream("stars/on.png")));
+				OFF_STAR = new ImageIcon(ImageIO.read(ConfigPanel.class.getResourceAsStream("stars/off.png")));
+			}
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Creates a new {@code PluginListItem} for a plugin.
+	 * <p>
+	 * Note that {@code config} and {@code configDescriptor} can be {@code null}
+	 * if there is no configuration associated with the plugin.
+	 */
+	PluginListItem(ConfigPanel configPanel, Plugin plugin, PluginDescriptor descriptor,
+		@Nullable Config config, @Nullable ConfigDescriptor configDescriptor)
+	{
+		this(configPanel, plugin, config, configDescriptor,
+			descriptor.name(), descriptor.description(), descriptor.tags());
+	}
+
+	/**
+	 * Creates a new {@code PluginListItem} for a core configuration.
+	 */
+	PluginListItem(ConfigPanel configPanel, Config config, ConfigDescriptor configDescriptor,
+		String name, String description, String... tags)
+	{
+		this(configPanel, null, config, configDescriptor, name, description, tags);
+	}
+
+	private PluginListItem(ConfigPanel configPanel, @Nullable Plugin plugin, @Nullable Config config,
+		@Nullable ConfigDescriptor configDescriptor, String name, String description, String... tags)
+	{
+		this.configPanel = configPanel;
+		this.plugin = plugin;
+		this.config = config;
+		this.configDescriptor = configDescriptor;
+		this.name = name;
+		this.description = description;
+		Collections.addAll(keywords, name.toLowerCase().split(" "));
+		Collections.addAll(keywords, tags);
```

Comment:
`keywords.addAll(tags)`, etc

Reply:
`List#addAll` accepts a `Collection`, not a `String` array. :(

Reply:
`Arrays.asList()`